src/server/src/jobs/ActivationReminderJob.js

```diff
@@ -0,0 +1,85 @@
+import cron from 'node-cron';
+import Mailing from '../../emailController'
+import User from "../models/user";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+dotenv.config();
+import logger from '../../logger';
+import { generateNewNotActivatedEmail } from "../resources/emails";
+
+/*
+* Returns users that didnt activate their account
+*/
+const getNewNotActivated = async () => {
+    const users = await User.find({activated: false});
+    logger.info(users);
+    return users;
+};
+
+/*
+* delete all users that didnt activate after DAYS_DELETE_UNACTIVED_USERS
+*/
+const removeOldNotActivated = async () => {
+    let now = new Date(); // get current time
+
+    //set min time
+    let min = new Date(); 
+    const gracePeriod = 24*60*60 * 1000 * process.env.DAYS_DELETE_UNACTIVED_USERS
+    min.setTime(now.getTime() - gracePeriod);
+
+    // go over all unactivated users and delete all that didnt activate in the grace period
+    for await (const doc of User.find({creationDate: {$gt: min, $lt: now}})) {
```

Comment:
Maybe gather a list of all the users we wish to delete and execute the delete command in the DB level one time?

Reply:
done